lib/timers/promises.js

```diff
@@ -111,7 +115,70 @@ function setImmediate(value, options = {}) {
       () => signal.removeEventListener('abort', oncancel)) : ret;
 }
 
+async function* setInterval(after, value, options = {}) {
+  validateObject(options, 'options');
+  const { signal, ref = true, throwOnAbort = true } = options;
+  validateAbortSignal(signal, 'options.signal');
+  validateBoolean(ref, 'options.ref');
+  validateBoolean(throwOnAbort, 'options.throwOnAbort');
+
+  if (signal?.aborted) {
+    if (throwOnAbort) throw new AbortError();
+    return;
+  }
+
+  let onCancel;
+  let notYielded = 0;
+  let passCallback;
+  let abortCallback;
+  const interval = new Timeout(() => {
+    notYielded++;
+    if (passCallback) {
+      passCallback();
+      passCallback = undefined;
+      abortCallback = undefined;
+    }
+  }, after, undefined, true, true);
+  if (!ref) interval.unref();
+  insert(interval, interval._idleTimeout);
+  if (signal) {
+    onCancel = () => {
+      // eslint-disable-next-line no-undef
+      clearInterval(interval);
+      if (abortCallback) {
+        abortCallback(new AbortError());
+        passCallback = undefined;
+        abortCallback = undefined;
+      }
+    };
+    signal.addEventListener('abort', onCancel, { once: true });
+  }
+
+  while (!signal?.aborted) {
+    if (notYielded === 0) {
+      try {
+        await new Promise((resolve, reject) => {
+          passCallback = resolve;
+          abortCallback = reject;
```

Comment:
You don't actually need to save both resolve and reject, you can save resolve and pass `Promise.reject` to it. Not sure if it's worth it.

Reply:
done, I think less functions makes the code simpler.